### PR TITLE
Added ability to target mouse position

### PR DESCRIPTION
**Pull request by po-trottier**
Should be similar to Hades controls

- Character points towards where they're going
- When you attack, attack in the direction the mouse points in
- Stop the player from moving for a brief moment
- Return to the default behaviour

When attacking, we need to add a tiny delay to let the player model rotate in the proper direction. The delay is currently 0.1 sec which makes it barely noticeable! Could be pushed even lower (down to 0.02s)

### CODE REVIEW

File: Hush/Assets/Scripts/Weapon/PlayerSword.cs [hunk 1]
```diff
@@ -79,12 +93,22 @@ public float AttemptCrit(float damage)
...
+            playerAnimator.SetTrigger(PlayerAnimator.HeavyAttack);
+        }
+
         private void OnTriggerEnter(Collider col)
         {
             if (!col.CompareTag(Tags.Enemy) && !col.CompareTag(Tags.Dome)) 
                 return;
         
-            var stateInfo = playerAnimator.GetCurrentAnimatorStateInfo(PlayerAnimator.Layer.UpperBody);
+            var stateInfo = playerAnimator.GetCurrentAnimatorStateInfo((int)PlayerAnimator.Layer.UpperBody);
```
Comment (redknight990) [inline Hush/Assets/Scripts/Weapon/PlayerSword.cs:111]: setting the type of the Layer enum might save you the cast to int maybe, not sure exactly how it works in C#

File: Hush/Assets/Scripts/Player/PlayerMovement.cs [hunk 2]
```diff
@@ -75,13 +108,17 @@ private void MovePlayer()
             
             // Move the player
             controller.Move(transform.TransformDirection(horizontalDisplacement + verticalDisplacement));
-
-            // Update animator
+            
+            // Update the animator
             animator.SetFloat(PlayerAnimator.Speed, _playerSpeed);
+            Debug.Log(controller.velocity.magnitude);
```
Comment (redknight990) [inline Hush/Assets/Scripts/Player/PlayerMovement.cs:114]:
```suggestion
```
Remove debug log

**Review by redknight990 — DISMISSED** [review]
Clean! Looks good, just one comment and a trailing debug log

File: Hush/Assets/Scripts/Weapon/PlayerSword.cs (hunk 1 above)
Comment (po-trottier) [reply Hush/Assets/Scripts/Weapon/PlayerSword.cs:111]: it is set! (See Hush/Assets/Scripts/Player/Enums/PlayerAnimator.cs) it doesn't auto cast still!

**Review by redknight990 — APPROVED** [review]

**Pull request merged**